Website UI Updated to Reflect Blender UI

TuckleberryWinn opened the pull request.

TuckleberryWinn committed:
Separate assets, styles, and pages into their own folders.

TuckleberryWinn committed:
Start of Blender UI styling.

Organized directories
Radial gradient backgrounds
Navbar work to match Blender UI

TuckleberryWinn committed:
Page style update

-Added initial layout for learning and community tabs
-Updated navbar styling to be mobile first
-Added node editor style boxes for content sections
-Added templates for pages and graph nodes

TuckleberryWinn committed:
Add Blender SVG files

TuckleberryWinn committed:
Clean up navbar

Fix navbar positioning + add navbar decoration based on window width.
- Blender icon--> Secret Blender site link
- Shader node icon
- Object data node for pages
- Radio buttons for render mode

TuckleberryWinn committed:
Added goat display for viewport shading toggle in header.

TuckleberryWinn committed:
Add Gizmo toggle to Navbar in desktop view

TuckleberryWinn committed:
Update Header template across other pages

TuckleberryWinn committed:
Update header background to only scale at large resolution

AdmiralPotato commented on community/index.html:
```diff
@@ -0,0 +1,128 @@
+            id="shading-solid"
+            value="shading-solid"
+            name="shading-picker"
+          />
+          <input
+            type="radio"
+            value="shading-material"
+            id="shading-material"
+            name="shading-picker"
+          />
+          <input
+            type="radio"
+            value="shading-rendered"
+            id="shading-rendered"
+            name="shading-picker"
+          />
+          <div class="shading-toggle-buttons toolBar-button">
+            <label class="toolBar-shading-option" for="shading-wireframe">
+              <div class="toolBar-icon icon-image-shading-wireframe"></div>
+            </label>
+            <label class="toolBar-shading-option" for="shading-solid">
+              <div class="toolBar-icon icon-image-shading-solid"></div>
+            </label>
+            <label class="toolBar-shading-option" for="shading-material">
+              <div class="toolBar-icon icon-image-shading-material"></div>
+            </label>
+            <label class="toolBar-shading-option" for="shading-rendered">
+              <div class="toolBar-icon icon-image-shading-rendered"></div>
+            </label>
+            <div class="icon-down-arrow"></div>
+          </div>
+          <div id="rendered-goat-container">
+            <div class="rendered-goat-image viewport-goat-wireframe"></div>
+            <div class="rendered-goat-image viewport-goat-solid"></div>
+            <div class="rendered-goat-image viewport-goat-material"></div>
+            <div class="rendered-goat-image viewport-goat-rendered"></div>
+          </div>
+        </div>
+      </div>
+    </nav>
```
I really like what's being accomplished in the page with the code you've written here! I do have some recommendations on how this can be done a little simpler though.

- Try not to mix `camelCase` and `kabob-case` CSS class names. Pick one.
- I feel like a lot of things could have fewer total classes? Do something like putting a single class on the `<nav class="toolbar">`, and then write your CSS selectors like `.toolbar .icon`, to impact all icons inside the thing at the top level. Then like... most of these class names go away.

Fabulous work so far!!!

AdmiralPotato commented on community/index.html:
```diff
@@ -0,0 +1,128 @@
+        </div>
+      </div>
+    </nav>
+    <main>
+      <img src="../assets/Under_Construction.gif" style="width: 320px" />
+    </main>
```
Can you please put this into a node block so it gets centered?

AdmiralPotato commented on learning/index.html:
```diff
@@ -0,0 +1,143 @@
+      </div>
+    </nav>
+    <main>
+      <h2>Resources:</h2>
+      <ul>
+        <li>
+          <a href="https://blender.org/" target="_blank"
+            >https://blender.org/</a
+          >
+        </li>
+        <li>
+          <a
+            href="https://www.youtube.com/playlist?list=PLa1F2ddGya_-UvuAqHAksYnB0qL9yWDO6"
+            target="_blank"
+            >Blender Fundamentals video training playlist</a
+          >
+        </li>
+      </ul>
+      <img src="../assets/Under_Construction.gif" style="width: 320px" />
```
Can you please put these into a node block so it gets centered?

AdmiralPotato commented on styles.css:
```diff
@@ -0,0 +1,470 @@
+  display: block;
+}
+
+input[type="radio"],
+input[type="checkbox"] {
+  display: none;
+}
```
Can we prefix this effect with some kind of class or parent, like to only do this in certain parent chains? I think I actually prefer the idea of selecting these by their parent, rather than adding a class attribute on every input tag because that's smaller HTML.

AdmiralPotato commented on styles.css:
```diff
@@ -0,0 +1,470 @@
+  display: block;
+  color: #969696;
+  font-size: 1.2rem;
+  line-height: 1.6em;
+  background-color: #232323;
+  padding: 0 0.5em;
+  margin: 0.25em 0.2em 0;
+  border-radius: 0.75em 0.75em 0 0;
+}
+li.navBarTab.navBarCurrentPage a {
+  background-color: #303030;
+}
+.navBarTab.navBarCurrentPage a {
+  color: #f2f2f2;
+}
+@media (min-width: 768px) {
+  .navBar {
+    grid-template-columns: 1fr 400px 1fr;
+  }
+  .navDecoration {
+    display: flex;
+    align-items: center;
+    padding: 0.25rem;
+  }
+  .navBarTab a {
+    font-size: 1.5rem;
+  }
+}
+
+@media (min-width: 900px) {
+  .toolBar-button.extraDecoration {
+    display: flex;
+  }
+}
+@media (min-width: 1300px) {
+  .toolBar-shading-toggle,
+  .toolBar-gizmo-toggle {
+    display: flex;
+  }
+}
```
Same comment as in the HTML, don't do both `camelCase` and `kabob-case`, pick one.

AdmiralPotato commented on templates/graph_node:
Be sure to update this template after you redo some of the CSS above. :)

AdmiralPotato commented on templates/page_w_header:
Be sure to update this template after you redo some of the CSS above. :)

AdmiralPotato reviewed (changes requested):
Fabulous work! Some of the CSS and the number of class names on HTML elements can be simplified, but this is so awesome looking!

AdmiralPotato commented on assets/blender_viewport.webp:
Can you please replace this with the lossless one we made together?

TuckleberryWinn committed:
Addressing PR notes

-Updated class naming to be consistent kebab-case
-Consolidated some class names to reduce bloat
-Fixed double border on goat render reveal
-Updated templates to reflect current page state
-Added node boxes to the under construction pages

TuckleberryWinn committed:
The smallest nit

The tiniest div to span pipeline adjustment.
